NOJIRA cjs2esm

ragnarwestad opened the pull request:
Gå over til ES modules på hele prosjektet, inkludert generate-local-config og Vite config

Retter:
The CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

ragnarwestad committed:
Gå over til ES modules på hele prosjektet

Retter:
The CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

ragnarwestad committed:
Gå over til ES modules på hele prosjektet, inkludert generate-local-config og Vite config

Retter:
The CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

ragnarwestad committed:
Merge remote-tracking branch 'origin/NOJIRA_cjs2esm' into NOJIRA_cjs2esm

ragnarwestad committed:
Oppgradert import "uuid" til siste versjon.
Fjernet import av "@types/uuid" siden den nå er inkludert i "uuid"
Samkjørt bruk av uuid i koden slik at alle bruker src/utils importen.

Retter denne:
"npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details."

ragnarwestad committed:
Oppdatert Vite config til å bruke ESM syntaks

ragnarwestad committed:
Merge branch 'master' into NOJIRA_cjs2esm

# Conflicts:
#	package-lock.json

ragnarwestad committed:
Merge branch 'master' into NOJIRA_cjs2esm

# Conflicts:
#    package-lock.json

ragnarwestad committed:
Merge branch 'master' into NOJIRA_cjs2esm

# Conflicts:
#    .gitignore
#    package-lock.json
#    package.json

IsaSalemHame commented on nais/vars-q2.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "APP_NAME": "melosys-web",
+  "VITE_APP_NAME": "melosys-web",
```
Disse + nais-fila er ikke en del av frontend-bygg, og trenger ikke renames. Litt misvisende

ragnarwestad replied:
Good catch, revertert.

ragnarwestad committed:
Merge branch 'master' into NOJIRA_cjs2esm

ragnarwestad committed:
Revertert endring i APP_NAME

ragnarwestad committed:
- Tell Vite to treat `/env-config.mjs` as an external dependency that shouldn't be bundled, preventing import analysis issues
- Use proper ES module import syntax in index.html instead of classic script loading, enabling module-based loading

ragnarwestad committed:
Merge branch 'master' into NOJIRA_cjs2esm

ragnarwestad committed:
Revert "Oppgradert import "uuid" til siste versjon."

This reverts commit dbe7793092134ac05b727e854184a77632e08874.

ragnarwestad committed:
Revertert endring i APP_NAME

ragnarwestad committed:
Merge branch 'master' into NOJIRA_cjs2esm

ragnarwestad committed:
Oppdatert

ragnarwestad commented:
Dropper denne, ble for mye rot. Lager en clean ny
